### ISSUE TITLE
List page scrolling drift occurs occasionally

### ISSUE THREAD

**Issue by gadfly3173** [sourcegit-scm/sourcegit#126]
https://github.com/sourcegit-scm/sourcegit/assets/28685179/c7b2eb7e-70dc-4f25-ae1e-ea6bb4d7dd39

When I stay on the list page and switch directly to other applications, such as Chrome and IDEA, use the mouse wheel to scroll down a certain distance, switch back to SourceGit, and scroll again on the list page, the list page may suddenly scroll down for a large distance. distance. Currently only found on my Deepin OS 20.9, may be a Linux platform specific issue.

**Comment by love-linger** [comment]
I can not reproduce issue on my Debian 12 (Wayland)

**Comment by gadfly3173** [comment]
![image](https://github.com/sourcegit-scm/sourcegit/assets/28685179/5ca1c784-bf0c-4022-a93c-63ac1bb5c7af)
It seems to instantly pick up all the wheel actions I had while leaving the app when I switch back to SourceGit scrolling.

```bash
xdpyinfo
name of display:    :0
version number:    11.0
vendor string:    The X.Org Foundation
vendor release number:    12011000
X.Org version: 1.20.11
maximum request size:  16777212 bytes
motion buffer size:  256
bitmap unit, bit order, padding:    32, LSBFirst, 32
image byte order:    LSBFirst
number of supported pixmap formats:    7
supported pixmap formats:
    depth 1, bits_per_pixel 1, scanline_pad 32
    depth 4, bits_per_pixel 8, scanline_pad 32
    depth 8, bits_per_pixel 8, scanline_pad 32
    depth 15, bits_per_pixel 16, scanline_pad 32
    depth 16, bits_per_pixel 16, scanline_pad 32
    depth 24, bits_per_pixel 32, scanline_pad 32
    depth 32, bits_per_pixel 32, scanline_pad 32
keycode range:    minimum 8, maximum 255
focus:  window 0xca0000a, revert to PointerRoot
number of extensions:    28
    BIG-REQUESTS
    Composite
    DAMAGE
    DOUBLE-BUFFER
    DPMS
    DRI2
    DRI3
    GLX
    Generic Event Extension
    MIT-SCREEN-SAVER
    MIT-SHM
    Present
    RANDR
    RECORD
    RENDER
    SECURITY
    SHAPE
    SYNC
    X-Resource
    XC-MISC
    XFIXES
    XFree86-DGA
    XFree86-VidModeExtension
    XINERAMA
    XInputExtension
    XKEYBOARD
    XTEST
    XVideo
default screen number:    0
number of screens:    1
```

**Comment by love-linger** [comment]
I've switch my Debain from Wayland to X11. And I still can NOT reproduce this issue.

https://github.com/sourcegit-scm/sourcegit/assets/4075673/39f8f4b1-b96b-4990-a63d-4f4f643c506f

**Comment by gadfly3173** [comment]
I did not reproduce this problem when using other system tests in the virtual machine. It may be caused by some very complicated reasons.

**Comment by julaudo** [comment]
Reproduced this issue quite easily
Ubuntu 22.04.4 LTS
Gnome 42.9
Windowing system X11

**Comment by gadfly3173** [comment]
It seems fixed in Avalonia 11.1.2 https://github.com/AvaloniaUI/Avalonia/pull/16185
